src/js/views/right_panel_view.js

```diff
@@ -524,15 +522,25 @@
 
             // Render template for each position and append to $el
             var html = "";
+            var filteredLbls = []
             _.each(positions, function(lbls, p) {
-
                 lbls = _.map(lbls, function(label, key){ return label; });
 
                 var json = {'position':p, 'labels':lbls};
                 if (lbls.length === 0) return;
+                console.log(lbls)
+                lbls.forEach(function(lbl){
+                    filteredLbls.push(lbl)
+                })
+                
                 json.inner_template = self.inner_template;
                 html += self.template(json);
             });
+           
+            this.models.forEach(function(m){
+                m.save('labels', filteredLbls);
+            })
```

Comment:
I'm not sure what the changes in `right_panel_view.js` are doing (are they needed)?
But this seems to be breaking "undo" functionality since with every re-render you are saving a change to the model.

Reply:
It is to handle any label edition. For example, if you have two labels ``aa``, one green, one blue, both at the same size and position, then if you turn the green one to blue, only one blue label is remaining. But obviously, if it breaks the undo, I have to do it differently